src/providers/eip-1559-gas-price-provider.ts

```diff
@@ -43,14 +43,14 @@ export class EIP1559GasPriceProvider extends IGasPriceProvider {
     super();
   }
 
-  public async getGasPrice(): Promise<GasPrice> {
+  public override async getGasPrice(requestBlockNumber: number): Promise<GasPrice> {
     const feeHistoryRaw = (await this.provider.send('eth_feeHistory', [
       /**
        * @fix Use BigNumber.from(this.blocksToConsider).toHexString() after hardhat adds support
        * @see https://github.com/NomicFoundation/hardhat/issues/1585 .___.
        */
       BigNumber.from(this.blocksToConsider).toHexString().replace('0x0', '0x'),
-      'latest',
+      BigNumber.from(requestBlockNumber).toHexString().replace('0x0', '0x'),
```

Comment:
is there any explanation for the `replace` part?

Reply:
Yea there's some explanation inside https://github.com/NomicFoundation/hardhat/issues/1585:

https://github.com/NomicFoundation/hardhat/issues/1585#issuecomment-874235095